niceman/retrace/packagemanagers.py

```diff
@@ -253,14 +253,21 @@ def _create_package(self, pkgname):
         for v in pkg_info.versions:
             v_info = {"version": v.version}
             origins = []
-            for o in v.origins:
-                origin = {"component": o.component,
-                          "archive": o.archive,
-                          "origin": o.origin,
-                          "label": o.label,
-                          "site": o.site}
+            for (pf, _) in v._cand.file_list:
+                # Pull origin information from package file
+                origin = {"component": pf.component,
+                          "archive": pf.archive,
+                          "origin": pf.origin,
+                          "label": pf.label,
+                          "site": pf.site}
+                # Get the archive uri
+                indexfile = v.package._pcache._list.find_index(pf)
+                if indexfile:
+                    archive_uri = indexfile.archive_uri("")
+                    origin["archive_uri"] = archive_uri
                 # Get the release date
-                rdate = self._find_release_date(o.site, o.archive)
+                rdate = self._find_release_date(
+                    self._find_release_file(pf.filename))
                 if rdate:
                     origin["date"] = rdate
```

Comment:
Here's another case where populating an object differs from my mental picture of this as a configuration file I'm generating.  I was avoiding populating "date" if it didn't exist so it wouldn't appear in the configuration file.  self._find_release_date returns "None" if not found, so the whole thing could simply become:

`origin["date"] = self._find_release_date(self._find_release_file(pf.filename))`

(Which I should just break down and do.  However, my hope is to create an Origin object next...)